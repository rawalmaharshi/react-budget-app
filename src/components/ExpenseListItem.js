import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

class ExpenseListItem extends React.Component {
  render() {
    return (
      <Link
        className="list-item"
        to={`/edit/${this.props.expense.id}`}>
        <div>
          <h3 className="list-item__title">{this.props.expense.description}</h3>
          <span className="list-item__sub-title">{moment(this.props.expense.createdAt).format('MMM Do, YYYY')}</span>
        </div>
        <h3 className="list-item__data">{numeral(this.props.expense.amount / 100).format('$0,0.00')} </h3>
      </Link>
    );
  }
}

export default ExpenseListItem;