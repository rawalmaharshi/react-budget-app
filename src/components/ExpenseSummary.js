import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpense from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => (
  <div>
    {
      props.expensesCount > 1 ?
        (
          <h2>Viewing {props.expensesCount} expenses totaling: {numeral(props.expensesTotal / 100).format('$0,0.00')} </h2>
        ) :
        (
          <h2>Viewing {props.expensesCount} expense totaling: {numeral(props.expensesTotal / 100).format('$0,0.00')} </h2>
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpense(state.expenses, state.filters);

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: expensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);