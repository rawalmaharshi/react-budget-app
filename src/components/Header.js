import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => (
  <div>
    <h1>Expensify</h1>
    <p>
      <NavLink activeClassName="is-active" to="/dashboard" exact={true}>Go Home</NavLink> --
      <NavLink activeClassName="is-active" to="/create">Add</NavLink> --
      <button onClick={props.startLogout}>Logout</button>
    </p>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);