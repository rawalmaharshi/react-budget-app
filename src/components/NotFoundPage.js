import React from 'react';

const NotFoundPage = () => (
  <div>
    This is 404!
  </div>
);

export default NotFoundPage;