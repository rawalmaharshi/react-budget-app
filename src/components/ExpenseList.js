import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpense from '../selectors/expenses';

export const ExpenseList = (props) => {
  return (
    <div className="content-container">
      <div className="list-header">
        <div className="show-for-mobile">Expenses</div>
        <div className="show-for-desktop">Expense</div>
        <div className="show-for-desktop">Amount</div>
      </div>
      <div className="list-body">
        {
          props.expenses.length === 0 ?
            (
              <div>
                <span className="list-item list-item__message">
                  No expenses
              </span>
              </div>
            ) :
            (
              props.expenses.map((expense) => (
                <ExpenseListItem
                  key={expense.id}
                  expense={expense}
                />
              ))
            )
        }
      </div>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpense(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);