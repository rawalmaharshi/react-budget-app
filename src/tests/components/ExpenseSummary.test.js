import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Should render ExpenseSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={125.56} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpenseSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={10} expensesTotal={5125.56} />);
  expect(wrapper).toMatchSnapshot();
});