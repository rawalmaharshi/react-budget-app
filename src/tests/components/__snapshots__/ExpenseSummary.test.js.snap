// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    10
     expenses totaling: 
    $51.26
     
  </h2>
</div>
`;

exports[`Should render ExpenseSummary correctly with one expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     expense totaling: 
    $1.26
     
  </h2>
</div>
`;
